snapcraft/cli/lifecycle.py

```diff
@@ -31,6 +40,9 @@ def _execute(command, parts, **kwargs):
     else:
         try:
             lifecycle.execute(command, project_options, parts)
+        except errors.SnapcraftUserPermissionError as user_error:
+            if user_error.user == 'root':
+                _reexec_as_root()
```

Comment:
@ogra1 so does build not actually need this anymore? I'm leaning more and more toward having an environment variable or build attribute that says prime needs to be done with sudo rather than automatically trying it.